test/unit/middleware/EvidenceUpload.test.js

```diff
@@ -0,0 +1,146 @@
+        }
+      }]}`);
+
+    beforeEach(function() {
+      this.timeout(2500);
+      stubs = {
+        formidable: {
+          IncomingForm: function() {
+            this.parse = parser;
+            this.once = () => {};
+            this.on = () => {};
+          }
+        },
+        request: {
+          post: poster
+        },
+        fs: {
+          unlink: unlinker,
+          createReadStream: () => {}
+        },
+        path: {
+          resolve: () => 'a string'
+        }
+      };
+      EvidenceUpload = proxyquire('steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js', stubs);
+      EvidenceUpload.makeDir = sinon.stub().callsArg(1);
+    });
+
+    afterEach(() => {
+      parser.reset();
+      unlinker.reset();
+      poster.reset();
+    });
+    it('creates a directory, forwards the file to the api then deletes it', done => {
+      EvidenceUpload.handleUpload({
+        method: 'post'
+      }, {}, error => {
+        expect(EvidenceUpload.makeDir).to.have.been.called;
+        expect(parser).to.have.been.called;
+        expect(poster).to.have.been.called;
+        expect(unlinker).to.have.been.called;
+        expect(error).not.to.exist;
+        done();
+      });
+    });
+    it('if create directory fails, it invokes the callback with the error', done => {
+      EvidenceUpload.makeDir = sinon.stub().yields(new Error('Bad dir!'));
+      EvidenceUpload.handleUpload({
+        method: 'post'
+      }, {}, error => {
+        expect(EvidenceUpload.makeDir).to.have.been.called;
+        expect(parser).not.to.have.been.called;
+        expect(poster).not.to.have.been.called;
+        expect(unlinker).not.to.have.been.called;
+        expect(error).to.exist;
+        done();
+      });
+    });
+    it('if uploading fails, it invokes the callback with the error', done => {
+      parser = sinon.stub().yields(new Error('argh!'), [], {
+        uploadEv: {
+          name: 'giacomo'
+        }
+      });
+      EvidenceUpload.handleUpload({
+        method: 'post'
+      }, {}, error => {
+        expect(EvidenceUpload.makeDir).to.have.been.called;
+        expect(parser).to.have.been.called;
+        expect(poster).not.to.have.been.called;
+        expect(unlinker).not.to.have.been.called;
+        expect(error).to.exist;
+        done();
+      });
+    });
+    it('if req.method is not post, it doesn\'t do anything apart from just invoking the callback', done => {
+      EvidenceUpload.handleUpload({
+        method: 'get'
+      }, {}, error => {
+        expect(EvidenceUpload.makeDir).not.to.have.been.called;
+        expect(parser).not.to.have.been.called;
+        expect(poster).not.to.have.been.called;
+        expect(error).not.to.exist;
+        done();
+      });
+    });
+  });
+});
+/* eslint-enable init-declarations */
+/* eslint-enable no-shadow */
+/* eslint-enable max-len */
+/* eslint-enable no-empty-function */
+/* eslint-enable func-names */
+/* eslint-enable object-shorthand */
```

Comment:
Could you add some more tests for the other functions in this class. `path()`, `middleware()`, `form()`, `values()`, `next()`. As our coverage has gone down a bit.

Reply:
Sure!